include/mxnet/lib_api.h

```diff
@@ -277,6 +289,14 @@ struct MXTensor {
     return size;
   }
 
+  /*! \brief helper function to compare two MXTensors */
+  inline bool operator==(const MXTensor &oth) {
```

Comment:
It will be more consistent with MXNet NDArray API to use ‘IsSame’, since operator== is confusing between same object and same tensor content.

Reply:
Sorry that I did not read the previous review.

Reply:
writing a response together with @rondogency:
For clarity, here were checking the following items: dtype, version ID, shape, and the pointer address of the data in memory. If two tensors have the same data pointer then they will have the same data values also. So here the two tensors will have the same content and will have the same tensor attributes (shape, type, version ID) also. 

I dont think we need to be consistent with NDarray API here, since users writing custom ops wont necessarily be familiar with NDarray API. And here we're trying to make the right API for this use-case (custom operators). Our comparison API also checks the version number, and the NDarray API does not. So we're already diverging from being consistent with NDArray API. 

For the record, here is the similar NDarray API:
https://github.com/apache/incubator-mxnet/blob/55e222b8c97f99193f832f785cf210f876632add/include/mxnet/ndarray.h#L212-L217

Reply:
Thank you for the explanation in detail!
1. If two tensors have the same data pointer then they will have the same data values also.
It is right. However, if two tensors have the same data values, they may not have the same data pointer. I did a simple test on C++ vector.
```c++
#include <iostream>
#include <vector>
using namespace std;

int main() {
  vector<int> a{1,2,3};
  vector<int> b{1,2,3};
  cout << (a == b) << endl; // 1
  cout << (a.data() == b.data()) << endl; // 0
  return 0;
}
```

2. we don't need to be consistent with NDarray API here.
Agree : )

Reply:
Thanks @wkcn, if @rondogency agrees ill change it back to 'isSame'

Reply:
I agree with @wkcn on this. `isSame` is less confusing.

Reply:
ok I agree with changing it back to isSame, thanks for the inspection on c++ vector!